contracts/RouterV1.sol

```diff
@@ -14,7 +15,31 @@ import {IFactory} from "./Factory/IFactory.sol";
 /// @title RouterV1
 /// @notice Convenience contract for ampleforth geyser
 /// @dev Security contact: dev-support@ampleforth.org
-contract RouterV1 {
+contract RouterV1 is IERC721Receiver {
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes calldata
+    ) external pure override returns (bytes4) {
+        return IERC721Receiver.onERC721Received.selector;
+    }
+
+    function create2Vault(address vaultFactory, bytes32 salt) public returns (address vault) {
+        vault = IFactory(vaultFactory).create2("", salt);
+    }
+
+    function depositStake(
+        address geyser,
+        address vault,
+        uint256 amount,
+        bytes calldata permission
+    ) public {
+        address stakingToken = IGeyser(geyser).getGeyserData().stakingToken;
+        TransferHelper.safeTransferFrom(stakingToken, msg.sender, vault, amount);
+        IGeyser(geyser).stake(vault, amount, permission);
+    }
```

Comment:
For transparency: this contract function was created to have atomicity for the action of depositing + staking. Otherwise, the frontend would have to do depositing and staking via two separate contract calls.

Might be necessary to do something similar with unstake + withdraw.

Will bring this up to Ampleforth